components/raftstore/src/store/fsm/apply.rs

```diff
@@ -1323,6 +1346,10 @@ where
             }
         }
 
+        if should_write {
+            info!("persist data and apply state"; "region_id" => self.region_id(), "peer_id" => self.id(), "state" => ?self.apply_state);
+            ctx.commit(self);
```

Comment:
Should we move this into an outer function?

Reply:
Yes, otherwise the response won't be returned for the last command.